### PR TITLE
Remove alert `group` duplication 

### PR DESCRIPTION
**Pull request by rbarry82**
Include general catch-up and refactoring now that we've brought it back up to date with best practice.

This is a necessary PR for passing alert rules.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -6,15 +6,15 @@
 
 """Prometheus Scrape Configuration charm.
 
-The Prometheus Scrape Configuration is a charm that
-allows to override configurations for the scrape jobs
-generated by charms that provide relations using the
-'metrics-endpoint' relation.
+The Prometheus Scrape Configuration charm
+allows overriding configuration of scrape jobs
+from any metrics provider the provides these metrics
```
Comment (sed-i) [inline src/charm.py:11]:
```suggestion
from any metrics provider that provides these metrics
```

File: src/charm.py [hunk 2]
```diff
@@ -6,15 +6,15 @@
...
+through the 'metrics-endpoint' relation using the
+`prometheus_scrape` interface.
```
Comment (sed-i) [inline src/charm.py:13]:
```suggestion
through the 'metrics-endpoint' relation.
```

File: src/charm.py [hunk 3]
```diff
@@ -28,141 +28,136 @@ class PrometheusScrapeConfigCharm(CharmBase):
...
 
     def __init__(self, *args):
-        """Construct the charm, no arguments needed."""
+        """Construct the charm."""
         super().__init__(*args)
 
-        self._target_relation_name = "configurable-scrape-jobs"
-        self._prometheus_relation_name = "metrics-endpoint"
+        self._metrics_provider_relation_name = "configurable-scrape-jobs"
+        self._metrics_consumer_relation_name = "metrics-endpoint"
```
Comment (sed-i) [inline src/charm.py:42]: Convert to class variables?

File: src/charm.py [hunk 4]
```diff
@@ -28,141 +28,136 @@ class PrometheusScrapeConfigCharm(CharmBase):
...
-        for downstream_relation in downstream_relations:
-            downstream_relation.data[self.app]["scrape_jobs"] = scrape_jobs
-            downstream_relation.data[self.app]["alert_rules"] = alert_rules
-            logger.debug("Updated downstream %s", downstream_relation.app)
-
-    @property
-    def _config(self):
-        return {key.replace("-", "_"): value for key, value in self.model.config.items()}
+        metrics_consumer_relation.data[self.app]["scrape_jobs"] = scrape_jobs
+        metrics_consumer_relation.data[self.app]["alert_rules"] = alert_rules
```
Comment (sed-i) [inline src/charm.py:146]: Leader guard?

**Review by sed-i — APPROVED** [review]

File: src/charm.py (hunk 4 above)
Comment (balbirthomas) [reply src/charm.py:146]: yes. Thanks.

File: src/charm.py (hunk 3 above)
Comment (sed-i) [reply src/charm.py:41]: i.e. "static" variables. If you're so inclined, use ClassVar
https://docs.python.org/3/library/typing.html#typing.ClassVar

Comment (balbirthomas) [reply src/charm.py]: There is no real benefit to doing this right now. If unit tests need this in the future we can move it else where then. Leaving it in the most private scope possible when there is no specific use case for making it more visible would be the right thing to do.

File: src/charm.py (hunk 4 above)
Comment (balbirthomas) [reply src/charm.py:166]: done

File: src/charm.py (hunk 1 above)
Comment (balbirthomas) [reply src/charm.py]: done

File: src/charm.py (hunk 2 above)
Comment (balbirthomas) [reply src/charm.py:13]: I do want to make clear what the interface is, since that is more important.

**Review by balbirthomas — APPROVED** [review]

**Pull request merged**